src/components/BookDropDown/BookDropDown.js

```diff
@@ -0,0 +1,204 @@
+import Tabs from '@material-ui/core/Tabs';
+import Tab from '@material-ui/core/Tab';
+import Box from '@material-ui/core/Box';
+import { API } from '../../store/api';
+
+function TabPanel(props) {
+  const { children, value, index, ...other } = props;
+
+  return (
+    <div
+      role='tabpanel'
+      hidden={value !== index}
+      id={`simple-tabpanel-${index}`}
+      aria-labelledby={`simple-tab-${index}`}
+      {...other}
+    >
+      {value === index && <Box p={3}>{children}</Box>}
+    </div>
+  );
+}
+
+TabPanel.propTypes = {
+  children: PropTypes.node,
+  index: PropTypes.any.isRequired,
+  value: PropTypes.any.isRequired,
+};
+const useStyles = makeStyles((theme) => ({
+  bookButton: {
+    padding: '0 8px',
+    width: 71,
+  },
+  bookCard: {
+    minWidth: 600,
+  },
+  chapterButton: {
+    padding: 0,
+    width: 36,
+    minWidth: 'unset',
+  },
+}));
+
+export default function BookDropDown(props) {
+  const classes = useStyles();
+  const { value, onChange, width, source } = props;
+  const [booksValue, setBooksValue] = useState([]);
+  const [open, setOpen] = React.useState(false);
+  const [book, setBook] = React.useState([]);
+  const [chapter, setChapter] = React.useState(1);
+
+  useEffect(() => {
+    if (source !== undefined) {
+      API.get(`/bibles/${source.sourceName}/versification`)
+        .then((response) => {
+          console.log(response.data.maxVerses);
+          setBooksValue(response.data.maxVerses);
+        })
+        .catch((err) => console.log(err));
+    }
+  }, [source]);
+
+  const closeBookListing = () => {
+    onChange({ book, chapter });
+    setOpen(false);
+  };
+  const [tabValue, setTabValue] = React.useState(0);
+
+  const handleChange = (event, newValue) => {
+    setTabValue(newValue);
+  };
+  const bookClicked = (book) => {
+    setBook(book);
+    setTabValue(1);
+  };
+  const chapterClicked = (chapter) => {
+    setChapter(chapter);
+    onChange({ book, chapter });
+    setOpen(false);
+  };
+
+  const bookName = value.book;
+  const chapterNumber = value.chapter;
+  useEffect(() => {
+    if (bookName) {
+      setBook(bookName);
+    }
+  }, [bookName]);
+
+  useEffect(() => {
+    if (chapterNumber) {
+      setChapter(chapterNumber);
+    }
+  }, [chapterNumber]);
+
+  const displayBooks = () => {
+    if (Object.keys(booksValue).length > 0) {
+      return Object.keys(booksValue).map((bookObject, i) => {
+        const book = bookObject;
+        return (
+          <Grid item xs={2} key={i}>
+            <Button
+              size='small'
+              variant='outlined'
+              className={classes.bookButton}
+              onClick={() => bookClicked(book)}
+            >
+              <span style={{ fontSize: 15 }}>{bookObject}</span>
+            </Button>
+          </Grid>
+        );
+      });
+    } else {
+      return (
+        <Grid item xs={4}>
+          <Typography gutterBottom variant='h10'>
+            No Source selected
```

Comment:
This case means that there are no books uploaded in the selected source/project.